### PR TITLE
fix(#358): search in process definitions and dmn resource definitions

### LINKED ISSUE #358: Filtering by the process name does not work in  the community UI
### Description

If I enter a search term in the instance overview, I do not see the instances filtered by this search term, instead I still see all instances

### Steps to Reproduce

- deploy a few process definitons with different names
- navigate to processes - instances
- enter a search team, e.g. "pivo"

Image

### Environment

win11

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by triskill**
closes #358

### CODE REVIEW

File: internal/sql/queries/process_definition.sql [hunk 1]
```diff
@@ -14,6 +14,7 @@ WHERE
 -- force sqlc to keep sort param
   CAST(sqlc.narg('sort') AS TEXT) IS CAST(sqlc.narg('sort') AS TEXT)
   AND (CAST(sqlc.narg('bpmn_process_id_filter') AS TEXT) IS NULL OR pd.bpmn_process_id = CAST(sqlc.narg('bpmn_process_id_filter') AS TEXT))
+  AND (CAST(sqlc.narg('search') AS TEXT) IS NULL OR pd.bpmn_process_id LIKE '%' || CAST(sqlc.narg('search') AS TEXT) || '%' OR pd.bpmn_process_name LIKE '%' || CAST(sqlc.narg('search') AS TEXT) || '%')
```
[the remark this answers is not in the document]
Comment (triskill) [reply internal/sql/queries/process_definition.sql]: Ok, i amended this change.

File: internal/sql/queries/process_definition.sql [hunk 2]
```diff
@@ -14,6 +14,9 @@ WHERE
 -- force sqlc to keep sort param
   CAST(sqlc.narg('sort') AS TEXT) IS CAST(sqlc.narg('sort') AS TEXT)
   AND (CAST(sqlc.narg('bpmn_process_id_filter') AS TEXT) IS NULL OR pd.bpmn_process_id = CAST(sqlc.narg('bpmn_process_id_filter') AS TEXT))
+  AND (CAST(sqlc.narg('search') AS TEXT) IS NULL
+           OR lower(pd.bpmn_process_id) LIKE '%' || lower(CAST(sqlc.narg('search') AS TEXT)) || '%'
```
Comment (Kuzna) [inline internal/sql/queries/process_definition.sql]: I'd rather use the same approach as in dmn_resource_definition.sql. Also if the client sends ?search= (empty string, not omitted), the param is non-null and the query becomes LIKE '%%' — i.e., matches everything but still pays the scan cost. Consider normalizing empty strings to nil in the REST layer (server.go:170, server.go:644) or adding AND length(@search) > 0 to the predicate.

File: internal/sql/queries/process_definition.sql [hunk 3]
```diff
@@ -14,6 +14,9 @@ WHERE
...
+           OR lower(pd.bpmn_process_name) LIKE '%' || lower(CAST(sqlc.narg('search') AS TEXT)) || '%')
```
Comment (Kuzna) [inline internal/sql/queries/process_definition.sql]: I'd rather use the same approach as in dmn_resource_definition.sql. Also if the client sends ?search= (empty string, not omitted), the param is non-null and the query becomes LIKE '%%' — i.e., matches everything but still pays the scan cost. Consider normalizing empty strings to nil in the REST layer (server.go:170, server.go:644) or adding AND length(@search) > 0 to the predicate.

File: internal/sql/queries/dmn_resource_definition.sql [hunk 4]
```diff
@@ -59,6 +59,13 @@ WHERE
     ELSE
         1
     END
+    AND
+    CASE WHEN @search IS NOT NULL THEN
+         lower(drd.dmn_resource_definition_id) like concat('%', lower(@search), '%')
+         OR lower(drd.dmn_definition_name) like concat('%', lower(@search), '%')
```
Comment (Kuzna) [inline internal/sql/queries/dmn_resource_definition.sql]: If the client sends ?search= (empty string, not omitted), the param is non-null and the query becomes LIKE '%%' — i.e., matches everything but still pays the scan cost. Consider normalizing empty strings to nil in the REST layer (server.go:170, server.go:644) or adding AND length(@search) > 0 to the predicate.

File: test/e2e/process_definition_test.go [hunk 5]
```diff
@@ -100,6 +100,18 @@ func TestRestApiProcessDefinition(t *testing.T) {
 		assert.Equal(t, 1, response.JSON200.TotalCount)
 	})
 
+	t.Run("test search filter", func(t *testing.T) {
```
Comment (Kuzna) [inline test/e2e/process_definition_test.go:103]: Missing test for case-insensitivity (e.g. searching V2 should still match service-task-input-output-v2).
Missing test for search returning multiple results or no results.
Missing test for empty string
Test only searches by name ("v2"). It doesn't assert that searching by bpmn_process_id also matches, even though the SQL does OR lower(pd.bpmn_process_id) LIKE ....

File: test/e2e/decision_definition_test.go [hunk 6]
```diff
@@ -141,6 +141,24 @@ func TestGetDmnResourceDefinitions(t *testing.T) {
 		assert.Equal(t, "name12", processInstances.JSON200.Items[1].DmnDefinitionName)
 		assert.Equal(t, "defId1", processInstances.JSON200.Items[1].DmnResourceDefinitionId)
 	})
+
+	t.Run("find dmn resource definition by search across id and name", func(t *testing.T) {
```
Comment (Kuzna) [inline test/e2e/decision_definition_test.go:145]: Missing test for case-insensitivity (e.g. searching V2 should still match service-task-input-output-v2).
Missing test for search returning multiple results or no results.
Missing test for empty string.

**Review by Kuzna — CHANGES_REQUESTED** [review]

File: internal/sql/queries/process_definition.sql (hunk 2 above)
Comment (triskill) [reply internal/sql/queries/process_definition.sql]: Ok, resolved.

File: internal/sql/queries/dmn_resource_definition.sql (hunk 4 above)
Comment (triskill) [reply internal/sql/queries/dmn_resource_definition.sql]: Ok, resolved.

File: test/e2e/process_definition_test.go (hunk 5 above)
Comment (triskill) [reply test/e2e/process_definition_test.go:103]: Ok, resolved.

File: internal/sql/queries/process_definition.sql (hunk 3 above)
Comment (triskill) [reply internal/sql/queries/process_definition.sql]: Ok, resolved.

File: test/e2e/decision_definition_test.go (hunk 6 above)
Comment (triskill) [reply test/e2e/decision_definition_test.go:145]: Ok, resolved.

File: test/e2e/process_definition_test.go [hunk 7]
```diff
@@ -100,6 +100,95 @@ func TestRestApiProcessDefinition(t *testing.T) {
...
+				Search: &search,
+			})
+			assert.NoError(t, err)
+			for _, item := range response.JSON200.Items {
+				assert.True(t, strings.Contains(
+					strings.ToLower(item.BpmnProcessId),
+					strings.ToLower(search),
+				))
+			}
+			//for _, item := range response.JSON200.Items {
```
Comment (Kuzna) [inline test/e2e/process_definition_test.go]: Please remove this commented-out code.

Comment (triskill) [reply test/e2e/process_definition_test.go]: Thanks, i removed it.

**Review by Kuzna — APPROVED** [review]
